rewrite-java/src/main/java/org/openrewrite/java/ChangeType.java

```diff
@@ -505,6 +511,18 @@ private Expression updateOuterClassTypes(Expression typeTree) {
         private boolean isTargetFullyQualifiedType(JavaType.@Nullable FullyQualified fq) {
             return fq != null && TypeUtils.isOfClassType(fq, originalType.getFullyQualifiedName()) && targetType instanceof JavaType.FullyQualified;
         }
+
+        private boolean becomesAmbiguous(JavaSourceFile cu) {
+            JavaType.FullyQualified newType = TypeUtils.asFullyQualified(targetType);
+            JavaType.FullyQualified oldType = TypeUtils.asFullyQualified(originalType);
+
+            return newType != null && oldType != null && cu.getImports().stream().anyMatch(imprt -> {
```

Comment:
Try to avoid using the streams API, even when convenient, but especially on paths that might get evaluated often; Adding imports is one of those hot paths where we really want to avoid these object allocations where we can.